src/test/java/uk/gov/pay/adminusers/resources/InviteResourceGenerateOtpIT.java

```diff
@@ -81,6 +80,36 @@ public void generateOtp_shouldSucceed_forServiceInvite_evenWhenTokenIsExpired_si
                 .statusCode(OK.getStatusCode());
     }
 
+    @Test
+    public void generateOtp_shouldSucceed_forServiceInvite_whenNoFieldsPresent() throws Exception {
+        givenAnExistingServiceInvite();
+        Map<Object, Object> invitationRequest = emptyMap();
+
+        givenSetup()
+                .when()
+                .body(mapper.writeValueAsString(invitationRequest))
+                .contentType(ContentType.JSON)
+                .post(format(INVITES_GENERATE_OTP_RESOURCE_URL, code))
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    @Test
+    public void generateOtp_shouldSucceed_forServiceInvite_whenPasswordAndPhoneNumberPresent() throws Exception {
+        givenAnExistingServiceInvite();
+        Map<Object, Object> invitationRequest = Map.of(
+                "telephone_number", TELEPHONE_NUMBER,
+                "password", PASSWORD);
+
+        givenSetup()
+                .when()
+                .body(mapper.writeValueAsString(invitationRequest))
+                .contentType(ContentType.JSON)
+                .post(format(INVITES_GENERATE_OTP_RESOURCE_URL, code))
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
```

Comment:
It does seem a bit strange to be sending a tel no and password as part of a otp generation request, but without more context I don't know what the other options are, so I trust this is sensible

Reply:
It is a bit strange but it’s consistent with how the `UserOtpDispatcher` works. It’s kind of reasonable to set the phone number when sending the OTP but the fact the password is in there is just because the behaviour on adminsers is very tied to the flow in selfservice. But we find the world as it is, not as we would like it to be.